src/util/expr_iterator.h

```diff
@@ -152,27 +165,39 @@ class depth_iterator_baset
   depth_iterator_baset &operator=(depth_iterator_baset &&other)
   { m_stack=std::move(other.m_stack); }
 
-  /// Obtain non-const exprt reference. Performs a copy-on-write on
-  /// the root node as a side effect. To be called only if a the root
-  /// is a non-const exprt. Do not call on the end() iterator
+  const exprt &get_root()
+  {
+    return m_stack.front().expr.get();
+  }
+
+  /// Obtain non-const exprt reference. Performs a copy-on-write on the root
+  /// node as a side effect.
+  /// \remarks
+  ///   To be called only if a the root is a non-const exprt.
+  ///   Do not call on the end() iterator
   exprt &mutate()
   {
     PRECONDITION(!m_stack.empty());
-    auto expr=std::ref(const_cast<exprt &>(m_stack.front().expr.get()));
+    // Cast the root expr to non-const
+    exprt *expr = &const_cast<exprt &>(m_stack.front().expr.get());
```

Comment:
Should this involve a call to `get_root()` ?

Reply:
Sure, since we have that function now we may as well use it here too. Applied.